docs/proposals/edi.md

```diff
@@ -0,0 +1,259 @@
+file, which is usually shared by multiple ESL mappings.
+
+```
+bal edi eslToBal <ESL file path or directory> <ESL segment definitions path> <output json path or directory>
+```
+
+If a directory containing multiple ESL files is given as the input, all ESLs will be converted to Ballerina mappings
+and written into the output directory.
+
+## Code generation
+
+Ballerina records can be generated using the EDI mapping file to represent an EDI document. There are two options to
+generate records in ballerina.
+
+- **Using ballerina command**
+
+  Ballerina records for the above EDI mapping in edi-mapping1.json can be generated as follows (generated Ballerina
+  records will be saved in orderRecords.bal).
+
+```shell
+bal edi gen edi-mapping1.json orderRecords.bal
+```
+
+Generated Ballerina records for the above mapping are shown below.
+
+```ballerina
+type Header_Type record {|
+   string orderId?;
+   string organization?;
+   string date?;
+|};
+
+type Items_Type record {|
+   string item?;
+   int quantity?;
+|};
+
+type SimpleOrder record {|
+   Header_Type header;
+   Items_Type[] items?;
+|};
+```
+
+- **Using compiler plugin at the build time**
+
+  The ballerina compiler plugin can be configured to generate EDI records during the build time. This support for ballerina is
+  still under development and once it is available, EDI records can be generated using that feature.
+
+## Parsing EDI files
+
+The below code reads the edi-sample1.edi into a json variable named "orderData" and then convert the orderData json to the
+generated record "SimpleOrder". Once EDI documents are mapped to the SimpleOrder record, any attribute in the EDI can be
+accessed using record's fields as shown in the example code below.
+
+```ballerina
+import ballerina/io;
+import chathurace/edi.core as edi;
+
+public function main() returns error? {
+    edi:EDIMapping mapping = check edi:readMappingFromFile("resources/edi-mapping1.json");
+
+    string ediText = check io:fileReadString("resources/edi-sample1.edi");
+    json orderData = check edi:readEDIAsJson(ediText, mapping);
+    io:println(orderData.toJsonString());
+
+    SimpleOrder order1 = check orderData.cloneWithType(SimpleOrder);
+    io:println(order1.header.date);
+}
+```
+
+"orderData" json variable value will be as follows (i.e. output of io:println(orderData.toJsonString())):
+```json
+{
+  "header": {
+    "orderId": "ORDER_1201",
+    "organization": "ABC_Store",
+    "date": "2008-01-01"
+  },
+  "items": [
+    {
+      "item": "A-250",
+      "quantity": 12
+    },
+    {
+      "item": "A-45",
+      "quantity": 100
+    },
+    {
+      "item": "D-10",
+      "quantity": 58
+    },
+    {
+      "item": "K-80",
+      "quantity": 250
+    },
+    {
+      "item": "T-46",
+      "quantity": 28
+    }
+  ]
+}
+```
+
+A sample Ballerina project which uses the EDI library is given
+in [here](https://github.com/chathurace/ballerina-edi/tree/main/samples/simpleEDI)
```

Comment:
```suggestion
in [here](https://github.com/chathurace/ballerina-edi/tree/main/samples/simpleEDI)
```
This seems to be in a personal GitHub repo. Please check.

Reply:
yes, the plan is to move them to this repo. once those are moved, I'll update this document.